core/src/main/java/com/airbnb/aerosolve/core/models/KDTreeModel.java

```diff
@@ -44,6 +45,12 @@ public int leaf(double x, double y) {
     }
   }
 
+  // assume kdtree node x is lat, and y is lng, return distance in miles between kdtree's max and min
+  public double miles(int id) {
```

Comment:
kdTree may not be only used in the lat, lng case, the distance unit may not always be mile. You may want to rename it to something more general like `distance` instead of `mile`